docs/articles/reference/migration.md

```diff
@@ -6,6 +6,181 @@ Bytewax version to the next. For a detailed list of all changes, see
+from datetime import datetime, timezone
+
+now = datetime.now(timezone.utc)
+```
+
+If you need the previously scheduled awake time, store it in an instance variable before returning it from {py:obj}`~bytewax.operators.UnaryLogic.notify_at`. Your design probably already has that stored in an instance variable.
+
+Before:
+
+```python doctest:SKIP
+from bytewax.inputs import DynamicSource
+
+
+class PeriodicSource(DynamicSource):
+    def build(self, now: datetime, worker_index: int, worker_count: int):
+        pass
+```
+
+```python doctest:SKIP
+from bytewax.inputs import DynamicSource
+
+
+class PeriodicSource(DynamicSource):
+    def build(self, step_id: str, worker_index: int, worker_count: int):
+        pass
+```
+
+### Standardization on Confluent's Kafka Serialization Interface
+
+Bytewax's bespoke Kafka schema registry and serialization interface has been removed in favor of using {py:obj}`confluent_kafka.schema_registry.SchemaRegistryClient` and {py:obj}`confluent_kafka.serialization.Deserializer`s and {py:obj}`confluent_kafka.serialization.Serializer`s directly. This now gives you direct control over all of the configuration options for serialization and supports the full range of use cases. Bytewax's Kafka serialization operators in {py:obj}`bytewax.connectors.kafka.operators` now take these Confluent types.
+
+#### With Confluent Schema Registry
+
+If you are using Confluent's schema registry (with it's magic byte prefix), you can pass serializers like {py:obj}`confluent_kafka.schema_registry.avro.AvroDeserializer` directly to our operators. See Confluent's documentation for all the options here.
+
+Before:
+
+```python doctest:SKIP
+from bytewax.connectors.kafka import operators as kop
+from bytewax.connectors.kafka.registry import ConfluentSchemaRegistry
+
+sr_conf = {"url": CONFLUENT_URL, "basic.auth.user.info": CONFLUENT_USERINFO}
+registry = ConfluentSchemaRegistry(SchemaRegistryClient(sr_conf))
+
+key_de = AvroDeserializer(client)
+val_de = AvroDeserializer(client)
+
+# Deserialize both key and value
+msgs = kop.deserialize("de", kinp.oks, key_deserializer=key_de, val_deserializer=val_de)
+```
+
+After:
+
+```python doctest:SKIP
+from bytewax.connectors.kafka import operators as kop
+from confluent_kafka.schema_registry import SchemaRegistryClient
+from confluent_kafka.schema_registry.avro import AvroDeserializer
+
+# Confluent's SchemaRegistryClient
+client = SchemaRegistryClient(
+    {"url": CONFLUENT_URL, "basic.auth.user.info": CONFLUENT_USERINFO}
+)
+key_de = AvroDeserializer(client)
+val_de = AvroDeserializer(client)
+
+# Deserialize both key and value
+msgs = kop.deserialize("de", kinp.oks, key_deserializer=key_de, val_deserializer=val_de)
+```
+
+
+#### With Redpanda Schema Registry
+
+If you are using Redpanda's schema registry or another setup for which the serialized form does not have any magic bytes and is the payload directly, we provide compatible serializer classes in {py:obj}`bytewax.connectors.kafka.serde`.
```

Comment:
Given the compatibility between the registries, it's not so much about which schema registry you are using, but which library you use for (de)serialization.
So rather than mentioning redpanda's schema registry here, I'd focus on the distinction between using `confluent_kafka` library for (de)serialization vs `fastavro` or any other library in any other language that doesn't use confluent's wire format.